tests/exporter/test_ingestexportservice.py

```diff
@@ -468,6 +469,73 @@ def test_add_links_no_duplicates(self):
         self.assertTrue(links.get_links()[0] == mock_link)
         self.assertTrue(links.get_links()[1] == another_mock_link)
 
+    SUBMISSION = {
+        "triggersAnalysis": False,
+        "_links": {
+            "self": {
+                "href": "http://api.ingest.data.humancellatlas.org/SubmissionEnvelope/1234"
+            }
+        }
+    }
+
+    def test_upload_error_posted_to_ingest_api(self):
+        with self.assertRaises(Exception):
+            # given:
+            exporter = IngestExporter(ingest_api=self.mock_ingest_api, dss_api=self.mock_dss_api,
+                                      staging_api=self.mock_staging_api)
+            # and:
+            self.mock_ingest_api.get_entity_by_uuid = MagicMock(return_value=self.SUBMISSION)
+            exporter.logger.info = MagicMock()
+            exporter.get_all_process_info = MagicMock()
+            exporter.get_metadata_by_type = MagicMock()
+            exporter.prepare_metadata_files = MagicMock()
+            exporter.bundle_links = MagicMock()
+            exporter.create_bundle_manifest = MagicMock()
+
+            error = Exception('Error thrown for Unit Test')
+            error_json = ExporterError(str(error)).getJSON()
+
+            exporter.upload_metadata_files = MagicMock(side_effect=error)
+
+            # when:
+            exporter.export_bundle(bundle_uuid=None, bundle_version=None, submission_uuid=None, process_uuid=None)
+
+        # then:
+        self.mock_ingest_api.create_submission_error.assert_called_once_with(
+            self.SUBMISSION.get("_links").get("self").get("href"),
+            error_json
+        )
+
+    def test_dss_upload_error_posted_to_ingest_api(self):
+        with self.assertRaises(Exception):
+            # given:
+            exporter = IngestExporter(ingest_api=self.mock_ingest_api, dss_api=self.mock_dss_api,
+                                      staging_api=self.mock_staging_api)
+            # and:
+            self.mock_ingest_api.get_entity_by_uuid = MagicMock(return_value=self.SUBMISSION)
+            exporter.logger.info = MagicMock()
+            exporter.get_all_process_info = MagicMock()
+            exporter.get_metadata_by_type = MagicMock()
+            exporter.prepare_metadata_files = MagicMock()
+            exporter.bundle_links = MagicMock()
+            exporter.create_bundle_manifest = MagicMock()
+            exporter.upload_metadata_files = MagicMock()
+            exporter.get_metadata_files = MagicMock(return_value=list())
+            exporter.get_data_files = MagicMock(return_value=list())
+
+            error = Exception('Error thrown for Unit Test')
+            error_json = ExporterError(str(error)).getJSON()
+            exporter.put_bundle_in_dss = MagicMock(side_effect=error)
+
+            # when:
+            exporter.export_bundle(bundle_uuid=None, bundle_version=None, submission_uuid=None, process_uuid=None)
```

Comment:
Here you could do 

```export_attempt = lamda: exporter.export_bundle(bundle_uuid=None, bundle_version=None, submission_uuid=None, process_uuid=None)
self.assertRaises(export_attempt, Exception)
```

This would save the need for the `with()` block above, potentially better readability